MonkeyLoader.Resonite.Integration/Configuration/SettingsDataFeedInjector.cs

```diff
@@ -108,16 +111,34 @@ protected override bool OnEngineReady()
 
         private static void EnsureColorXTemplate(DataFeedItemMapper mapper)
         {
-            if (!mapper.Mappings.Any((DataFeedItemMapper.ItemMapping mapping) => mapping.MatchingType == typeof(DataFeedValueField<colorX>)))
+            if (!_colorXTemplateCleanupDone)
+            {
+                Logger.Info(() => "Looking for previously injected colorX templates.");
+                foreach (var mapping in mapper.Mappings.Where((DataFeedItemMapper.ItemMapping mapping) => mapping.MatchingType == typeof(DataFeedValueField<colorX>) && mapping.Template.Target != null && mapping.Template.Target.Slot.Name == InjectedColorXTemplateName).ToList())
```

Comment:
This could just be `.Where(mapping => [TypeCheck] && mapping.Template.Target?.Slot.Name == InjectedColorXTemplateName)`, no need to ToList() it.

Reply:
I think I need to ToList() it because it calls mapper.Mappings.Remove(mapping); which modifies the collection that is being enumerated.

Reply:
Ah yes, correct. I'd just use ToArray then since you don't need to add anything to it 🤔